governance/program/tests/program_test/mod.rs

```diff
@@ -0,0 +1,547 @@
+use borsh::BorshDeserialize;
+use solana_program::{
+    instruction::Instruction,
+    program_error::ProgramError,
+    program_pack::{IsInitialized, Pack},
+    pubkey::Pubkey,
+    rent::Rent,
+    system_instruction,
+};
+
+use solana_program_test::ProgramTest;
+use solana_program_test::*;
+
+use solana_sdk::{
+    signature::{Keypair, Signer},
+    transaction::Transaction,
+};
+use spl_governance::{
+    instruction::{
+        create_realm, deposit_governing_tokens, set_vote_authority, withdraw_governing_tokens,
+    },
+    processor::process_instruction,
+    state::{
+        enums::{GovernanceAccountType, GoverningTokenType},
+        realm::{get_governing_token_holding_address, get_realm_address, Realm},
+        voter_record::{get_voter_record_address, VoterRecord},
+    },
+};
+
+pub mod cookies;
+use self::cookies::{RealmCookie, VoterRecordCookie};
+
+pub mod tools;
+use self::tools::map_transaction_error;
+
+pub struct GovernanceProgramTest {
+    pub banks_client: BanksClient,
+    pub payer: Keypair,
+    pub rent: Rent,
+}
+
+impl GovernanceProgramTest {
+    pub async fn start_new() -> Self {
+        let program_test = ProgramTest::new(
+            "spl_governance",
+            spl_governance::id(),
+            processor!(process_instruction),
+        );
+
+        let (mut banks_client, payer, _) = program_test.start().await;
+
+        let rent = banks_client.get_rent().await.unwrap();
+
+        Self {
+            banks_client,
+            payer,
+            rent,
+        }
+    }
+
+    pub async fn process_transaction(
+        &mut self,
+        instructions: &[Instruction],
+        signers: Option<&[&Keypair]>,
+    ) -> Result<(), ProgramError> {
+        let mut transaction =
+            Transaction::new_with_payer(&instructions, Some(&self.payer.pubkey()));
+
+        let mut all_signers = vec![&self.payer];
+
+        if let Some(signers) = signers {
+            all_signers.extend_from_slice(signers);
+        }
+
+        let recent_blockhash = self.banks_client.get_recent_blockhash().await.unwrap();
+
+        transaction.sign(&all_signers, recent_blockhash);
+
+        self.banks_client
+            .process_transaction(transaction)
+            .await
+            .map_err(map_transaction_error)
+    }
+
+    pub async fn get_account<T: BorshDeserialize>(&mut self, address: &Pubkey) -> T {
+        let raw_account = self
+            .banks_client
+            .get_account(*address)
+            .await
+            .unwrap()
+            .expect("GET-TEST-ACCOUNT-ERROR: Account not found");
+
+        T::try_from_slice(&raw_account.data).unwrap()
+    }
+
+    #[allow(dead_code)]
```

Comment:
Is this still necessary? Looks like all of these Cookie creators are used

Reply:
Yes it's needed. If you run all testes it's ok but if you run individual tests only then you get tons of warnings without those annotations.